lib/util/export/wholeExportDeclaration.ts

```diff
@@ -0,0 +1,43 @@
+import ts from 'typescript';
+
+/**
+ * Whole export when the file is found within the destFiles
+ */
+export type FileFound = {
```

Comment:
In this kind of case what I usually is 

```
type Base = { ... };
type SubCase1 = Base & { ... };
type SubCase2 = Base & { ... };
type Whole = SubCase1 | SubCase2;
```

Because `Base` isn't actually a type you can use as it's not really a `WholeExportDeclaration`, which has either `file: string` or `file: null`, WDYT?